couchbase/src/tests/collection.rs

```diff
@@ -0,0 +1,723 @@
+                    1,
+                    Some(MutationToken::new(1, 1, 1, BUCKET.to_string())),
+                )));
+            }
+            ()
+        });
+    let mocked_collection = Collection::new(
+        Arc::new(mock_core),
+        NAME.to_string(),
+        SCOPE.to_string(),
+        BUCKET.to_string(),
+    );
+    let result: CouchbaseResult<MutationResult> = mocked_collection
+        .remove(
+            key,
+            RemoveOptions::default().durability(DurabilityLevel::ClientVerified(
+                ClientVerifiedDurability::default().persist_to(PersistTo::One),
+            )),
+        )
+        .await;
+    assert!(result.is_err());
+}
+
+#[tokio::test]
+async fn get_touch_works() {
+    let key = Uuid::new_v4().to_string();
+    let (sender, _) = oneshot::channel();
+    let request = Request::Touch(TouchRequest {
+        id: key.clone(),
+        sender,
+        bucket: BUCKET.to_string(),
+        options: TouchOptions::default(),
+        scope: SCOPE.to_string(),
+        collection: NAME.to_string(),
+        expiry: Duration::from_secs(10),
+    });
+
+    let mut mock_core = MockCore::default();
+    mock_core
+        .expect_send()
+        .with(eq(request))
+        .times(1)
+        .returning(|x| {
+            if let Request::Touch(r) = x {
+                let _ = r.sender.send(Ok(MutationResult::new(
+                    1,
+                    Some(MutationToken::new(1, 1, 1, BUCKET.to_string())),
+                )));
+            }
+            ()
+        });
+    let mocked_collection = Collection::new(
+        Arc::new(mock_core),
+        NAME.to_string(),
+        SCOPE.to_string(),
+        BUCKET.to_string(),
+    );
+    let result: CouchbaseResult<MutationResult> = mocked_collection
+        .touch(key, Duration::from_secs(10), TouchOptions::default())
+        .await;
+    assert_eq!(result.unwrap().cas(), 1);
+}
+
+#[tokio::test]
+async fn get_unlock_works() {
+    let key = Uuid::new_v4().to_string();
+    let (sender, _) = oneshot::channel();
+    let request = Request::Unlock(UnlockRequest {
+        id: key.clone(),
+        sender,
+        bucket: BUCKET.to_string(),
+        options: UnlockOptions::default(),
+        scope: SCOPE.to_string(),
+        collection: NAME.to_string(),
+        cas: 1,
+    });
+
+    let mut mock_core = MockCore::default();
+    mock_core
+        .expect_send()
+        .with(eq(request))
+        .times(1)
+        .returning(|x| {
+            if let Request::Unlock(r) = x {
+                let _ = r.sender.send(Ok(()));
+            }
+            ()
+        });
+    let mocked_collection = Collection::new(
+        Arc::new(mock_core),
+        NAME.to_string(),
+        SCOPE.to_string(),
+        BUCKET.to_string(),
+    );
+    let result: CouchbaseResult<()> = mocked_collection
+        .unlock(key, 1, UnlockOptions::default())
+        .await;
+
+    assert!(result.is_ok());
+}
+
+#[tokio::test]
+async fn lookup_in_works() {
```

Comment:
Is there much value to these tests that just check that the Cas value is 0? Relatedly, the Cas value should never be 0 on a successful response from the server.

Reply:
I have just corrected it. Please review